backend/app/services/metrics_service.py

```diff
@@ -90,6 +91,39 @@ def get_latency_measurements(self, model_id: UUID, time_window: int = 60) -> lis
                     result.append(entry["latency_ms"])
             return result
 
+    def get_throughput_measurements(self, model_id: UUID, time_window: int = 60) -> tuple[int, list[tuple[float, int]]]:
+        """
+        Retrieve throughput measurements for a specific model within the given time window.
+
+        Args:
+            model_id: UUID of the model to filter measurements
+            time_window: Time window in seconds to look back for measurements (default 60s)
+
+        Returns: Tuple of (total_requests, list of (timestamp, inference_count) per second)
+        """
+        cutoff_time = datetime.now(UTC).timestamp() - time_window
+        with self._lock:
+            arr = self._array.copy()
```

Comment:
It's safe to perform calculations outside of the lock as we are working with the copy:

```python
with self._lock:
  arr = self._array.copy()

calculations_on_array(arr)
```